### DISCUSSION TITLE
Expand cx() functionality

### DISCUSSION THREAD

**Discussion by MarkusWendorf** [joe-bell/cva#100] [Ideas]
Hey there,

I really like the features that `cva` provides but for simple cases (where I have to concatenate classes), I used to use `clsx`.

For the most part `cx` works fine, but I do miss the `clsx({ "some-class-to-apply": someBoolean })`.

Are there plans to expand on that? Or maybe even just re-export clsx?

**Comment by joe-bell** [comment]
I’ve been toying the idea of allowing people to swap out the “cx” function with their own one, is that something you’d be interested in?

**Reply by MarkusWendorf** [reply]
@joe-bell not really, I wanted a slightly better out of the box experience of cva.

I can't see what other use cases one could cover by swapping out the cx function, other than what `clsx` provides.

Given that `clsx` is very small ([that's the whole code for it](https://github.com/lukeed/clsx/blob/master/src/index.js), I would directly include it.

**Reply by joe-bell** [reply]
I’ll think about it

**Reply by MarkusWendorf** [reply]
@joe-bell I think your idea for swapping out the cx functionality could be worth it. Looking at https://github.com/joe-bell/cva/discussions/141 it might be useful to be able to customise how classes are concatenated.

Not sure how the API should look like, but my first instinct would be to have a config function which creates a customised `cva` function. Which would then be consumed instead of the library `cva` function

```typescript
import { config } from "cva";
import "clsx" from "clsx";
export const cva = config({ cx: clsx });
```

**Reply by linden-dg** [reply]
Following up on this one, it would be great to be able to extend/wrap the `cx` functionality. I want to be able to pass through a custom `class`/`className` to be able to override properties - which `cva` supports already with e.g. `buttonVariants({ theme, className})`. But because of the way tailwind works, you need to use something like tailwind-merge to prevent conflicts. 

The order of the return function for `cva` is already perfect for tailwind-merge - with the `props?.className` coming after the base & variants. But without being able to wrap `cva`, I have to remember to wrap every single use of a variant in `twMerge`. It would be much better if I could extend the `cva` function e.g:


```ts
import { config } from "cva";
import { clsx, type ClassValue } from "clsx";
import { twMerge } from "tailwind-merge";

export const cva = config({
  cx: (...inputs: ClassValue[]) => twMerge(clsx(inputs)),
});
```

**Comment by joe-bell** [comment]
Hello! Just to let you know that this will be available in `cva@1.0`. Thanks for the suggestion!
